### ISSUE TITLE
Open file through NSApplication delegate

### ISSUE THREAD

**Issue by andydotxyz** [glfw/glfw#1458]
I want to support opening files dragged onto the application icon on Mac OS.
I have set up the bundle Info.plist to respond to this - but it is failing due to the application not handing [NSApplication OpenFile:].

Can you provide a hint on how I register a delegate or provide an implementation that should allow this to happen? The GLFW code provides an application and a delegate which I cannot seem to override.

Any hints appreciated, thanks.

**Comment by elmindreda** [comment]
I don't think there is a way to achieve this without modifying GLFW yet, but will look into supporting it at some point after 3.3 is out. I would prefer to avoid adding a macOS-specific GLFW event, however.

**Comment by andydotxyz** [comment]
I agree with your aim there. Perhaps there is a generic "file open request" or similar that could be triggered by a macOS open parameter or a drop on the icon or a drag and drop into the app on other operating systems?

I am implementing an abstraction at a higher level so the details are not particularly important.

Is there any workaround that I could use at the moment such as providing GLFW an alternative macOS app container?

**Comment by elmindreda** [comment]
I can't think of a good workaround that doesn't involve modifying the code of GLFW and at that point just add whatever temporary solution you wish.

Since we have platform specific init hints now, how about one for macOS that allows GLFW to pass these paths to the existing drop callback but with a `NULL` window argument?

**Comment by qno** [comment]
Is this connected to this issue [Opening files associated with a GLFW program after launch](https://github.com/glfw/glfw/issues/1024) ?

**Comment by andydotxyz** [comment]
I suspect that we could see a solution address both of these yes. Is it possible to handle it using filepath drop generic callback for all OS?
